bot.py

```diff
@@ -403,6 +404,137 @@ async def display_current_list_of_summoners(ctx):
         await ctx.send(embed=create_embed(embed_data))
 
 
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements
+@bot.command(name="remove", help="Remove player(s) from the list")
+async def remove_summoner(ctx, *, message):
+    """Remove summoner(s) from list
+    and send  the list to the bot"""
+
+    try:
+        # typing indicator
+        async with ctx.typing():
+            await asyncio.sleep(1)
+
+        # converting the message into list of summoners
+        summoner_to_remove_input = message.split(",")
+
+        # Exception case: attempt to remove more than 10 players
+        if len(summoner_to_remove_input) > MAX_NUM_PLAYERS_TEAM:
+            raise Exception (
+                "Limit Exceeded",
+                "You tried to remove more than 10 summoners! \
+                \nPlease remove {0} less summoners or consider using `clear` command".format(
+                    MAX_NUM_PLAYERS_TEAM - len(summoner_to_remove_input)
+                )
+            )
+
+        # Exception case: data/data.json file does not exist
+        if not os.path.exists(json_path):
+            raise Exception (
+                "Limit Exceeded",
+                "There is no summoner(s) added in the game.\nPlease add summoner(s) first!"
+            )
+
+        # for importing data from json file
+        file_data = ""
+        # initializing server id to a variable
+        server_id = str(ctx.guild.id)
+
+        if os.path.getsize(json_path) > 0:
+            with open(json_path, "r") as file:
+                file_data = json.load(file)
+        else:
+            raise Exception (
+                "Limit Exceeded",
+                "There is no summoner(s) added in the game.\nPlease add summoner(s) first!"
+            )
+
+        unmatched_summoner_name = []
+        if server_id in file_data:
+            for player_name in summoner_to_remove_input:
+                # pylint: disable=cell-var-from-loop
+                matched_summoner = pydash.find(file_data[server_id],
+                    lambda x: normalize_name(x["user_name"])==normalize_name(player_name))
+                if pydash.predicates.is_empty(matched_summoner):
+                    unmatched_summoner_name.append(player_name)
+                else:
+                    file_data[server_id].remove(matched_summoner)
+                    matched_summoner["user_name_input"]=player_name
```

Comment:
you are removing the names from data and then throwing exception if there is `unmatched_summoner_name` ->
This is going to remove the names but then error out if there is a single unmatched data no?

The saved list should not be touched at all if we are going to throw exception. 

List should be updated **after** all checks are completed

Reply:
Add `TODO: refactor` I'll work on it. This is taking too long now.